holopy/scattering/theory/scatteringtheory.py

```diff
@@ -277,3 +278,17 @@ def _transform_to_desired_coordinates(cls, detector, origin, wavevec=1):
             cls.desired_coordinate_system)
         return method(original_coordinate_values)
 
+
+def _select_scatterer(scatterer, illum):
```

Comment:
Since this is just for multi-wavelength, can we rename this to `_select_multiwavelength_scatterer` or something?

Also, what was happening before -- was there a bug that you caught or is this a new need?

Reply:
renamed to `select_scatterer_by_illumination`. In theory it should work for e.g. cross-polarized light sources as well as different wavelengths. 

This was previously handled by the Scatterer itself, but has been moved here as part of simplifying the Scatterer class.